src/server/routes/parcel.routes.ts

```diff
@@ -19,6 +21,48 @@ export default function registerRoutes(
     },
   );
 
+  async function getDeserializedParcel(payload: Buffer, logger: FastifyBaseLogger) {
+    try {
+      return await Parcel.deserialize(bufferToArrayBuffer(payload));
+    } catch (err) {
+      // Don't log the full error because 99.99% of the time the reason will suffice.
+      logger.info({ reason: (err as Error).message }, 'Refusing malformed parcel');
+      return null;
+    }
+  }
+
+  async function isMessageValid(parcel: Parcel, serviceOptions: ServiceOptions) {
+    try {
+      await serviceOptions.activeEndpoint.validateMessage(parcel);
+      return true;
+    } catch (err) {
+      serviceOptions.logger.info({ err }, 'Refusing invalid parcel');
+      return false;
+    }
+  }
+
+  async function getDecryptedParcel(parcel: Parcel, serviceOptions: ServiceOptions) {
+    try {
+      return await parcel.unwrapPayload(serviceOptions.activeEndpoint.keyStores.privateKeyStore);
+    } catch (err) {
+      serviceOptions.logger.info({ err }, 'Ignoring invalid service message');
+      return null;
+    }
+  }
+
+  async function storePda(pdaBuffer: Buffer, serviceOptions: ServiceOptions) {
+    try {
+      const privateEndpointConnParams = await PrivateEndpointConnParams.deserialize(pdaBuffer);
+      await serviceOptions.activeEndpoint.savePrivateEndpointChannel(
+        privateEndpointConnParams,
+        fastify.mongoose,
+      );
```

Comment:
`try` blocks should only handle one thing. Note I keep saying this.

In this particular case, the problem is that when there's an error, we log the message "Refusing to store invalid endpoint connection params". But how do you know that was actually the reason? How do you know the error didn't come from `activeEndpoint.savePrivateEndpointChannel()`?